aws/src/services/S3Storage.ts

```diff
@@ -53,7 +53,7 @@ export class S3Storage implements CloudStorage {
       Bucket: opts.container,
       Key: opts.path,
       Body: streamBody,
-      ContentLength: streamBody.readableLength
+      ContentLength: streamBody.readableLength || (streamBody as any)._readableState.length
```

Comment:
I don't like the idea of accessing private properties we may investigate further about this